public/shuffleFace.ts

```diff
@@ -0,0 +1,53 @@
+import { generate } from "../src/generate";
+import { raceBySkinColor } from "../src/globals";
+import {
+  FaceConfig,
+  GallerySectionConfig,
+  GenerateOptions,
+  Race,
+} from "../src/types";
+import { deepCopy, deleteFromDict } from "./utils";
+
+export const shuffleEntireFace = (
+  faceConfig: FaceConfig,
+  gallerySectionConfigList: GallerySectionConfig[],
+  setFaceStore: any,
+) => {
+  let faceConfigCopy = deepCopy(faceConfig);
+
+  let options: GenerateOptions = {};
+
+  for (let gallerySectionConfig of gallerySectionConfigList) {
+    if (gallerySectionConfig.randomizeEnabled) {
+      deleteFromDict(faceConfigCopy, gallerySectionConfig.key);
+    }
+
+    if (
+      !gallerySectionConfig.randomizeEnabled &&
+      gallerySectionConfig.key === "body.color" &&
+      faceConfig.body.color &&
+      raceBySkinColor[faceConfig.body.color]
+    ) {
+      let inferredRace: Race = raceBySkinColor[
+        faceConfig.body.color
+      ][0] as Race;
+      if (inferredRace) {
+        options.race = inferredRace;
```

Comment:
Not sure we want this, it should be an explicit feature to select the race.

Reply:
Should this be a feature we add to the FaceConfig? I know on previous site it wasn't added into Face type and just tracked via dropdown. Curious on your take for how it should be tracked (maybe Race as well?)

Reply:
Maybe, but for this PR I definitely don't want to change that.